homeassistant/components/light/__init__.py

```diff
@@ -369,3 +370,16 @@ def state_attributes(self):
     def supported_features(self):
         """Flag supported features."""
         return 0
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Component added, restore_state using platforms."""
+        state = get_last_state(self)
+        if state:
```

Comment:
Let's add another helper to this file that wraps the entity logic around `get_state`.

```python
def extract_info(state):
    """Restore properties from state object."""
    return {
        is_on: state.state == STATE_ON,
        # etc…
    }

class Light:
    def async_added_to_hass(self):
        restore_state.xxxxx(self, extract_info)
```

And in restore state, the helper `xxxxx` does all the logic that was in light before.

```python
def xxxxx(entity, extract_info):
    if not hasattr(entity, 'async_restore_state'):
        return

    state = get_state(entity.entity_id)

    if not state:
        return

    yield from entity.async_restore_state(**extract_info(state))
```